### ISSUE TITLE
Cope with recent `setuptools` changes to editable mode packaging (PEP 660)

### ISSUE THREAD

**Issue by jacobtylerwalls** [archesproject/arches#9937]
### Background ###
Versions of `setuptools` since 64.0.0 [changed](https://setuptools.pypa.io/en/latest/history.html#id168) editable (development) mode packaging with `-e`, especially (only?--not sure) with `pyproject.toml` based builds, see [PEP 660](https://peps.python.org/pep-0660/). Arches uses a pyproject.toml, apparently only to configure `black` (!), which might be the reason we're in this situation.

The setuptools changes are good, since it forces developers to test on systems set up more like their users' systems. In the meantime, though, this means that imports that were depending on the old `-e` behavior will fail with the new `-e` behavior.

For instance, multiple builds were failing yesterday on manage.py commands like so, suggesting that a relative import somewhere was being resolved to an ugly/nonexistent path like `arches.app.models.app.models.models...`:
```py
RuntimeError: Conflicting 'bulkindexqueue' models in application 'models': <class 'arches.app.models.models.BulkIndexQueue'> and <class 'arches.app.models.app.models.models.BulkIndexQueue'>.
```

### Action item ###
The stopgap fix in #9933 was to just not use `-e` on CI builds, since it's [better](https://blog.ganssle.io/articles/2019/08/test-as-installed.html) to test against the installed package than a source checkout.

The medium term workaround for developers who need editable mode is to opt-in to [legacy editable mode](https://setuptools.pypa.io/en/latest/userguide/development_mode.html#legacy-behavior):
```py
pip install --upgrade pip
pip uninstall arches
pip install -e . --config-settings editable_mode=compat
```

We could add that setting to our setup.py/pyproject.toml, or we could just fix the root cause of the incompatibilities with the new behavior (likely adjusting relative imports to be more explicit), since the "compat" mode is a transitional setting that will go away at some point. Opening this ticket to track that work.

**Comment by jacobtylerwalls** [comment]
> For instance, multiple builds were failing yesterday on manage.py commands like so, suggesting that a relative import somewhere was being resolved to an ugly/nonexistent path like arches.app.models.app.models.models...:

Turns out this "sudden" failure last month was a [bug](https://github.com/pypa/setuptools/issues/4019) in setuptools 68.1.0, fixed in 68.1.1. So maybe we aren't on the hook for fixing the "compat mode" issues...

**Comment by jacobtylerwalls** [comment]
Glancing over this again I don't think there's an action item. Editable installs are working fine with latest pip 'n setuptools.
